test/units/parsing/test_ajson.py

```diff
@@ -20,3 +24,147 @@ def test_AnsibleJSONDecoder_vault():
+
+        return M(request.param)
+
+    @pytest.fixture
+    def ansible_json_encoder(self):
+        return AnsibleJSONEncoder()
+
+    ###############
+    # Test methods:
+
+    @pytest.mark.parametrize(
+        'test_input,expected',
+        [
+            (datetime(2019, 5, 14, 13, 39, 38, 569047), '2019-05-14T13:39:38.569047'),
+            (datetime(2019, 5, 14, 13, 47, 16, 923866), '2019-05-14T13:47:16.923866'),
+            (date(2019, 5, 14), '2019-05-14'),
+            (date(2020, 5, 14), '2020-05-14'),
+        ]
+    )
+    def test_date_datetime(self, ansible_json_encoder, test_input, expected):
+        """
+        Test for passing datetime.date or datetime.datetime objects
+        to AnsibleJSONEncoder.default()
+        """
+        assert(ansible_json_encoder.default(test_input) == expected)
+
+    @pytest.mark.parametrize(
+        'mapping,expected',
+        [
+            ({1: 1}, {1: 1}),
+            ({2: 2}, {2: 2}),
+            ({1: 2}, {1: 2}),
+            ({2: 1}, {2: 1}),
+        ], indirect=['mapping']
+    )
+    def test_mapping(self, ansible_json_encoder, mapping, expected):
+        """
+        Test for passing Mapping object to AnsibleJSONEncoder.default()
+        """
+        m = mapping
+        assert(isinstance(m, Mapping))
+        assert(ansible_json_encoder.default(m) == expected)
+
+    @pytest.mark.parametrize(
+        'test_input,expected',
+        [
+            (vault_data()['data_0'], vault_data()['expected_0']),
+            (vault_data()['data_1'], vault_data()['expected_1']),
+        ]
+    )
+    def test_ansible_json_decoder_vault(self, ansible_json_encoder, test_input, expected):
+        """
+        Test for passing AnsibleVaultEncryptedUnicode to AnsibleJSONEncoder.default()
+        """
+        assert(isinstance(test_input, AnsibleVaultEncryptedUnicode))
+        assert(ansible_json_encoder.default(test_input) == {'__ansible_vault': expected})
+
+    @pytest.mark.parametrize(
+        'test_input,expected',
+        [
+            ({1: 'first'}, {1: 'first'}),
+            ({2: 'second'}, {2: 'second'}),
+        ]
+    )
+    def test_default_encoder(self, ansible_json_encoder, test_input, expected):
+        """
+        Test for the default encoder of json.JSONEncoder superclass
+        by passing objects of different classes that are not tested above
+        """
+        assert(ansible_json_encoder.default(test_input) == expected)
+
+    @pytest.mark.parametrize('test_input', [1, 1.1, 'string', [1, 2], set('set'), True, None])
+    def test_default_encoder_unserializable(self, ansible_json_encoder, test_input):
+        """
+        Test for the default encoder of json.JSONEncoder superclass
+        by passing unserializable objects of different classes.
+        It must fail with TypeError 'object is not serializable'
+        """
+        with pytest.raises(TypeError):
+            assert(ansible_json_encoder.default(test_input))
```

Comment:
If you expect this invocation to emit an exception, `assert`ion doesn't make sense here:
```suggestion
            ansible_json_encoder.default(test_input)
```